src/main/java/ua/tc/marketplace/service/CategoryService.java

```diff
@@ -1,8 +1,35 @@
 package ua.tc.marketplace.service;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import ua.tc.marketplace.model.dto.category.CreateCategoryDTO;
 import ua.tc.marketplace.model.entity.Category;
 
+/**
+ * Service interface defining operations for managing categories. Includes methods for
+ * retrieving, creating, updating, and deleting categories.
+ *
+ * <p>This interface provides operations for:
+ * <ul>
+ *   <li>Retrieving a category by its ID</li>
+ *   <li>Retrieving all categories with pagination support</li>
+ *   <li>Creating a new category</li>
+ *   <li>Updating an existing category</li>
+ *   <li>Deleting a category by its ID</li>
+ * </ul>
+ * </p>
+ */
 public interface CategoryService {
+    Category findCategoryById(Long categoryId);
 
-  Category findCategoryById(Long categoryId);
+    Page<CreateCategoryDTO> findAll(Pageable pageable);
```

Comment:
return type in this method should be just `CategoryDto` which contains all of the fields of Category

Reply:
It was you who created this method for AdServiceImpl